hack/update-bazel.sh

```diff
@@ -17,10 +17,23 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-if ! command -v bazel &>/dev/null; then
-  echo "Install bazel at https://bazel.build" >&2
-  exit 1
+
+if [ -x "$(command -v docker)" ]; then
+    CONTAINER_ENGINE=docker
+elif [ -x "$(command -v podman)" ]; then
+    CONTAINER_ENGINE=podman
+else
+	echo "There is no docker or podman installed. Please use hack/update-bazel.sh script instead."
```

Comment:
Ideally we just merge these two scripts, aka something along the lines of:

* If bazel or bazelisk is installed, use that
* Otherwise use an image.


```sh
bazel-from-image() {
  $CONTAINER_ENGINE run --user $(id -u) ... $@
}

bazel-direct() {
  bazel $@
}

bazel-from-bazelisk() {
  bazelisk $@
}


if [ -x $(command -v bazelisk) ]; then
  bazel=bazel-from-bazelisk
elif [ -x $(command -v bazel) ]; then
  bazel=bazel-direct
else
  bazel=bazel-from-image
fi

"$bazel" run @io_k8s_repo_infra//hack:update-bazel
```

Reply:
But what if a user wants to run from an image even if he has bazel already installed? That was my case at least. I guess adding an argument will be ok.

Reply:
@fejta updated.